src/main.go

```diff
@@ -40,23 +43,53 @@ func init() {
 
 	// 4. Seed random number generator
 	rand.Seed(time.Now().UnixNano())
+
+	cfg.K8sClient = cluster.ConnectK8sClient()
+	license.InitializeConfig(cfg.K8sClient)
 }
 
 // ========== //
 // == Main == //
 // ========== //
 
 func main() {
+
+	lis, server := CreateListenerAndGrpcServer()
+	// add license server
+	server = grpcserver.AddLicenseServer(server)
+
+	// check for license secret, if exist then validate
+	err := license.CheckLicenseSecret()
+
+	if err != nil {
+		log.Error().Msgf("error while validating license secrets for discovery engine, error: %s", err.Error())
+		go serve(lis, server)
+		_ = license.LCfg.WatchFeatures()
+		os.Exit(1)
```

Comment:
I  don't understand the point of spawning a go routine when that go routine would not be handled since the code is going to do os.Exit().